prism/src/main/java/org/apache/falcon/resource/extensions/ExtensionManager.java

```diff
@@ -0,0 +1,432 @@
+    }
+
+    @POST
+    @Path("suspend/{job-name}")
+    @Consumes({MediaType.TEXT_XML, MediaType.TEXT_PLAIN})
+    @Produces({MediaType.TEXT_XML, MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON})
+    public APIResult suspend(@PathParam("job-name") String jobName,
+                             @DefaultValue("") @QueryParam("doAs") String doAsUser) {
+        try {
+            List<Entity> entities = getEntityList("", "", "", TAG_PREFIX_EXTENSION_JOB + jobName, "", doAsUser);
+            for (Entity entity : entities) {
+                if (entity.getEntityType().isSchedulable()) {
+                    if (getWorkflowEngine(entity).isActive(entity)) {
+                        getWorkflowEngine(entity).suspend(entity);
+                    }
+                }
+            }
+        } catch (FalconException | IOException e) {
+            LOG.error("Error when scheduling extension job: " + jobName + ": ", e);
+            throw FalconWebException.newAPIException(e, Response.Status.BAD_REQUEST);
+        }
+        return new APIResult(APIResult.Status.SUCCEEDED, "Extension job " + jobName + " suspended successfully");
+    }
+
+    @POST
+    @Path("resume/{job-name}")
+    @Consumes({MediaType.TEXT_XML, MediaType.TEXT_PLAIN})
+    @Produces({MediaType.TEXT_XML, MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON})
+    public APIResult resume(@PathParam("job-name") String jobName,
+                            @DefaultValue("") @QueryParam("doAs") String doAsUser) {
+        try {
+            List<Entity> entities = getEntityList("", "", "", TAG_PREFIX_EXTENSION_JOB + jobName, "", doAsUser);
+            for (Entity entity : entities) {
+                if (entity.getEntityType().isSchedulable()) {
+                    if (getWorkflowEngine(entity).isSuspended(entity)) {
+                        getWorkflowEngine(entity).resume(entity);
+                    }
+                }
+            }
+        } catch (FalconException | IOException e) {
+            LOG.error("Error when resuming extension job " + jobName + ": ", e);
+            throw FalconWebException.newAPIException(e, Response.Status.BAD_REQUEST);
+        }
+        return new APIResult(APIResult.Status.SUCCEEDED, "Extension job " + jobName + " resumed successfully");
+    }
+
+    @POST
+    @Path("delete/{job-name}")
+    @Consumes({MediaType.TEXT_XML, MediaType.TEXT_PLAIN})
+    @Produces({MediaType.TEXT_XML, MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON})
+    public APIResult delete(@PathParam("job-name") String jobName,
+                            @DefaultValue("") @QueryParam("doAs") String doAsUser) {
+        try {
+            List<Entity> entities = getEntityList("", "", "", TAG_PREFIX_EXTENSION_JOB + jobName, "", doAsUser);
+            for (Entity entity : entities) {
+                // TODO(yzheng): need to remember the entity dependency graph for clean ordered removal
```

Comment:
This is for untrusted extensions where we can have multiple entities in one job. In this case, we need to know the dependency among entities to get the right order of deletion. See "Future Work" in design doc.